tests/integration_tests/db_engine_specs/elasticsearch_tests.py

```diff
@@ -0,0 +1,54 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from sqlalchemy import column
+
+from superset.db_engine_specs.elasticsearch import ElasticSearchEngineSpec
+from tests.integration_tests.db_engine_specs.base_tests import TestDbEngineSpec
+
+
+class TestElasticsearchDbEngineSpec(TestDbEngineSpec):
+    def test_timegrain_week_expression(self):
+        """
+        DB Eng Specs (elasticsearch): Test time grain expressions
+        """
+        col = column("ts")
+        test_cases = {
```

Comment:
The `parametrized` module (which mimics aspects of `pytest` is more preferable than using a for loop to iteratate over the test cases. Here's an [example](https://github.com/apache/superset/blob/be8265794059d8bbe216a4cb22c7a3f6adf4bcb3/tests/integration_tests/charts/api_tests.py#L1730-L1736).

Reply:
@john-bodley thanks for showing me this example. However, I noticed that I made a mistake before. There is only the [date type](https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-data-types.html#sql-data-types) that is supported in Elasticsearch SQL. I adjusted the tests on the `datetime` type only.